File: ReactWindows/ReactNative/Animation/AnimationRegistry.cs
using ReactNative.Bridge;
using System.Collections.Generic;

namespace ReactNative.Animation
{
    /// <summary>
    /// Coordinates catalyst animations driven by <see cref="UIManagerModule"/>.
    /// </summary>
    public class AnimationRegistry
    {
        private readonly IDictionary<int, AnimationManager> _animationRegistry;

        public AnimationRegistry() { }

        public AnimationRegistry(AnimationManager animation)
        {
            DispatcherHelpers.AssertOnDispatcher();

            _animationRegistry = new Dictionary<int, AnimationManager>() {
                { animation.AnimationId, animation}
            };
        }

        public AnimationManager GetAnimation(int animationID)
        {
            DispatcherHelpers.AssertOnDispatcher();
            var animation = default(AnimationManager);
            
            if (_animationRegistry.TryGetValue(animationID, out animation))
            {
                return animation;
            }
            else
            {
                return null;
            }
        }

        public void RegisterAnimation(AnimationManager animation)
        {
            DispatcherHelpers.AssertOnDispatcher();

            _animationRegistry[animation.AnimationId] = animation;
        }

        public AnimationManager RemoveAnimation(int animationID)
        {
            DispatcherHelpers.AssertOnDispatcher();

            var animation = default(AnimationManager);

            if (_animationRegistry.TryGetValue(animationID, out animation))
            {
                _animationRegistry.Remove(animationID);
            }

            return animation;
        }
    }
}

File: ReactAndroid/src/main/java/com/facebook/react/animation/AnimationRegistry.java
/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.animation;

import android.util.SparseArray;

import com.facebook.react.bridge.UiThreadUtil;

/**
 * Coordinates catalyst animations driven by {@link UIManagerModule} and
 * {@link AnimationManagerModule}
 */
public class AnimationRegistry {

  private final SparseArray<Animation> mRegistry = new SparseArray<Animation>();

  public void registerAnimation(Animation animation) {
    UiThreadUtil.assertOnUiThread();
    mRegistry.put(animation.getAnimationID(), animation);
  }

  public Animation getAnimation(int animationID) {
    UiThreadUtil.assertOnUiThread();
    return mRegistry.get(animationID);
  }

  public Animation removeAnimation(int animationID) {
    UiThreadUtil.assertOnUiThread();
    Animation animation = mRegistry.get(animationID);
    if (animation != null) {
      mRegistry.delete(animationID);
    }
    return animation;
  }

}
